packages/babel-generator/src/printer.js

```diff
@@ -584,15 +589,26 @@ export default class Printer {
       this._append(val);
     });
 
-    // Always add a newline after a block comment
-    this.newline(isBlockComment && !this._noLineTerminator ? 1 : 0);
+    if (printNewLines) this.newline(1);
   }
 
-  _printComments(comments?: Array<Object>) {
+  _printComments(comments?: Array<Object>, inlinePureAnnotation?: boolean) {
     if (!comments || !comments.length) return;
 
-    for (const comment of comments) {
-      this._printComment(comment);
+    if (
+      inlinePureAnnotation &&
+      comments.length === 1 &&
+      PURE_ANNOTATION_RE.test(comments[0].value)
+    ) {
+      this._printComment(
+        comments[0],
+        // Keep newlines if the comment marks a standalone call
+        this._buf.hasContent() && !this.endsWith("\n"),
```

Comment:
This is for cases like https://github.com/babel/babel/blob/master/packages/babel-plugin-proposal-async-generator-functions/test/fixtures/async-generators/expression/output.js#L1, where I think that we should keep the `\n`.